src/app/qgsclipboard.cpp

```diff
@@ -131,14 +133,30 @@ void QgsClipboard::generateClipboardText( QString &textContent, QString &htmlCon
 
         for ( int idx = 0; idx < attributes.count(); ++idx )
         {
-          QString value = attributes.at( idx ).toString();
+          QString value;
+          if ( attributes.at( idx ).canConvert( QVariant::String ) )
```

Comment:
canConvert is unfortunately not safe to use here -- the definition has changed in Qt6 from "i definitely CAN represent this value as a string" to "it's theoretical possible to create a string value here regardless of what the variant type is" . 

So for Qt 6 builds a list variant will return true for canConvert( QVariant::String ). 

I think you'll need to invert this logic, and have a hardcoded list of types which you want to format as json. I.e. QVariant::StringList, QVariant::List, QVariant::Map. And for the rest fallback to the toString() method.